EMS/EMS/SearchEngine.cpp

```diff
@@ -2,31 +2,78 @@
 
 list<Employee*> SearchEngine::search(list<Employee>& db, int employeeNum) const
 {
+	list<Employee*> result;
 
-	return list<Employee*>();
+	for (auto& item : db) {
+		if (item.employeeNum == employeeNum)
+			result.emplace_back(&item);
+	}
+	return result;
 }
 
 list<Employee*> SearchEngine::search(list<Employee>& db, Name& name) const
 {
-	return list<Employee*>();
+	list<Employee*> result;
+
+	for (auto& item : db) {
+		if (name.first != "" && name.first != item.name.first) continue;
+		if (name.last != "" && name.last != item.name.last) continue;
+
+		result.emplace_back(&item);
+	}
+
+	return result;
 }
 
-list<Employee*> SearchEngine::search(list<Employee>& db, CL& cl) const
+list<Employee*> SearchEngine::search(list<Employee>& db, CL cl) const
 {
-	return list<Employee*>();
+	list<Employee*> result;
+
+	for (auto& item : db) {
+		if (item.cl == cl)
+			result.emplace_back(&item);
+	}
+
+	return result;
 }
 
 list<Employee*> SearchEngine::search(list<Employee>& db, PhoneNum& phoneNum) const
 {
-	return list<Employee*>();
+	list<Employee*> result;
+
+	for (auto& item : db) {
+		if (phoneNum.mid != -1 && phoneNum.mid != item.phoneNum.mid) continue;
+		if (phoneNum.last != -1 && phoneNum.last != item.phoneNum.last) continue;
+
+		result.emplace_back(&item);
+	}
+
+	return result;
 }
 
 list<Employee*> SearchEngine::search(list<Employee>& db, Birthday& birthDay) const
 {
-	return list<Employee*>();
+	list<Employee*> result;
+
+	for (auto& item : db) {
+		if (birthDay.year != -1 && birthDay.year != item.birthDay.year) continue;
+		if (birthDay.month != -1 && birthDay.month != item.birthDay.month) continue;
+		if (birthDay.day != -1 && birthDay.day != item.birthDay.day) continue;
+
+		result.emplace_back(&item);
+	}
+
+	return result;
 }
 
-list<Employee*> SearchEngine::search(list<Employee>& db, CERTI& certi) const
+list<Employee*> SearchEngine::search(list<Employee>& db, CERTI certi) const
 {
-	return list<Employee*>();
+	list<Employee*> result;
+
+	for (auto& item : db) {
+		if (certi == item.certi)
+			result.emplace_back(&item);
```

Comment:
마찬가지로 들여쓰기를 줄이기 위해 제안드립니다.
```suggestion
		if (certi != item.certi) continue;
		result.emplace_back(&item);
```

Reply:
넵, 제안 감사합니다 :) 수정했습니다. 확인 부탁드려요 !